src/shmem_collectives.h

```diff
@@ -76,6 +76,10 @@ shmem_internal_sync(int PE_start, int logPE_stride, int PE_size, long *pSync)
         RAISE_ERROR_MSG("Illegal barrier/sync type (%d)\n",
                         shmem_internal_barrier_type);
     }
+
+    /* Ensure remote updates are visible in memory */
+    shmem_internal_membar_acquire();
```

Comment:
Should we instead call `membar_release`? We would like to ensure the stores become visible, right?

Reply:
That is a good question.  We put acquire fences in wait/test.  Maybe we actually need acq_rel?

Reply:
The acquire prevents operations below the fence from floating up.  But, I think we also want a release to prevent operations from before the fence falling below.

Reply:
Yes, acq_rel should be the safest option. But, if we want to optimize to either acquire or release, we should do release, I think.

Reply:
I don't think release is sufficient, it still allows the architecture to reorder a subsequent load to occur before the fence.

Reply:
Thanks for the careful review -- will fix this.  :)

Reply:
Yes, we might need both. Thanks.

Reply:
Fixed by 50e7890

Reply:
@jdinan This one still needs to be fixed.